consul/leader.go

```diff
@@ -76,6 +76,19 @@ func (s *Server) leaderLoop(stopCh chan struct{}) {
 	var reconcileCh chan serf.Member
 	establishedLeader := false
 
+	reassert := func() error {
```

Comment:
why move this away? I'd keep the logic where it is used unless you want to call `reassert` multiple times. Easier to read.